
When no SSID is set, the reconnect decision should entirely be left to
userspace. The driver should not decide which AP to associate with based on
arbitrary policies.

Signed-off-by: Samuel Ortiz <sameo@linux.intel.com>
---
 drivers/staging/rt2860/common/mlme.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: b/drivers/staging/rt2860/common/mlme.c
===================================================================
--- a/drivers/staging/rt2860/common/mlme.c	2010-01-05 11:25:22.000000000 +0100
+++ b/drivers/staging/rt2860/common/mlme.c	2010-01-05 13:10:32.000000000 +0100
@@ -1554,7 +1554,8 @@ void MlmeAutoReconnectLastSSID(struct rt
 	else if ((pAd->Mlme.CntlMachine.CurrState == CNTL_IDLE) &&
 		 (MlmeValidateSSID
 		  (pAd->MlmeAux.AutoReconnectSsid,
-		   pAd->MlmeAux.AutoReconnectSsidLen) == TRUE)) {
+		   pAd->MlmeAux.AutoReconnectSsidLen) == TRUE) &&
+		 (pAd->MlmeAux.AutoReconnectSsidLen != 0)) {
 		struct rt_ndis_802_11_ssid OidSsid;
 		OidSsid.SsidLength = pAd->MlmeAux.AutoReconnectSsidLen;
 		NdisMoveMemory(OidSsid.Ssid, pAd->MlmeAux.AutoReconnectSsid,
